### PR TITLE
Run go test in CI

### PR DESCRIPTION
**Pull request by tomleb**
# Issue https://github.com/rancher/rancher/issues/44474

The test script is not running in the `ci` script. 

The changes:
- Run the test script in the ci script
- Move dapper image to `v0.6.0` as that is the latest image (2 years old). The 1.11.2 version is 7 years old [according to docker hub](https://hub.docker.com/r/rancher/dapper/tags?page=1&name=1.11.2).

I have dropped the following changes:
- The git command added allows users on linux to run dapper with DAPPER_MODE=bind which makes builds a bit faster by not requiring the repository to be copied, a bind mount can be used instead.
  - I'm not sure that's the proper way to fix this, need more investigation from me.
- Add volumes for caching go modules. This also makes the build faster.
  - This caching might work differently on GHA so instead of making it work both here and later, I'll defer to when we migrate to GHA.
- Run CI on arm64 in drone: Let's not add anything in drone unless necessary.

### CODE REVIEW

File: scripts/ci [hunk 1]
```diff
@@ -5,3 +5,4 @@ cd $(dirname $0)
 
 ./build
 ./validate
+./test
```
Comment (pmatseykanets) [inline scripts/ci:8]: Should we change the order?
```suggestion
./validate
./test
./build
```

**Review by pmatseykanets — APPROVED** [review]

File: scripts/ci (hunk 1 above)
Comment (tomleb) [reply scripts/ci:8]: I'm okay with either. Anyone else wanna chime in on this or expand on which order would be best?

**Comment by tomleb** [comment]
I have updated the PR by removing the following:

> I have dropped the following changes:
> - The git command added allows users on linux to run dapper with DAPPER_MODE=bind which makes builds a bit faster by not requiring the repository to be copied, a bind mount can be used instead.
>  - I'm not sure that's the proper way to fix this, need more investigation from me.
>- Add volumes for caching go modules. This also makes the build faster.
>  - This caching might work differently on GHA so instead of making it work both here and later, I'll defer to when we migrate to GHA.
> - Run CI on arm64 in drone: Let's not add anything in drone unless necessary.

I have also changed the order of script called in script/ci.

File: scripts/ci (hunk 1 above)
Comment (tomleb) [reply scripts/ci]: Updated to your suggestion.

**Review by pmatseykanets — APPROVED** [review]

**Review by maxsokolovsky — APPROVED** [review]

**Pull request merged**